db_test.go

```diff
@@ -437,3 +437,89 @@ func assertContains[K fmt.Stringer](s *SurrealDBTestSuite, input []K, matcher fu
 	s.NotEmptyf(matching, "Input %+v did not contain matching element", fmt.Sprintf("%+v", input))
 	return matching
 }
+
+func (s *SurrealDBTestSuite) TestIsDuplicateUniqueIdx() {
+	query := `DEFINE INDEX uniqueNameIdx ON TABLE users COLUMNS username UNIQUE;`
+	_, err := s.db.Query(query, nil) // because of this query, I modified the 'username' field in other tests.
+	s.Require().NoError(err)
+
+	userNoSpecialChars := testUser{
+		Username: "johnny",
+		Password: "1234",
+	}
+	data, err := s.db.Create("users", userNoSpecialChars) // true
+	s.Require().NoError(err)
+	s.Require().NotNil(data) // we collect this result, so we can get the record it to use it for other test
+
+	var createdUser []testUser
+	s.Require().NoError(surrealdb.Unmarshal(data, &createdUser))
+	s.Require().NotEmpty(createdUser[0].ID)
+
+	userWithSpecialChars := testUser{
+		Username: "jim",
+		Password: "abcd",
+	}
+	_, err = s.db.Create("users", userWithSpecialChars) // true
+	s.Require().NoError(err)
+
+	userWithObjectID := map[string]interface{}{
+		"username": struct {
+			Field1 string
+		}{
+			Field1: "nestedID",
+		},
+		"password": "4321",
+	}
+	_, err = s.db.Create("users", userWithObjectID)
+	s.Require().NoError(err)
+
+	// now, the actual tests...
+	msg := "the operation must fail"
+	s.Run("normal characters work", func() {
+		_, err = s.db.Create("users", userNoSpecialChars)
+		s.Require().NotNil(err, msg) // this validation is redundant, the function already checks for nil, but just in case
+		s.True(surrealdb.IsDuplicateUniqueIdx(err))
+	})
+	err = nil
+
+	s.Run("special characters work", func() {
+		_, err = s.db.Create("users", userWithSpecialChars)
+		s.Require().NotNil(err, msg)
+		s.True(surrealdb.IsDuplicateUniqueIdx(err))
+	})
+	err = nil
+
+	s.Run("records with objects as id work", func() {
+		_, err = s.db.Create("users", userWithObjectID)
+		s.Require().NotNil(err, msg)
+		s.True(surrealdb.IsDuplicateUniqueIdx(err))
+	})
+	err = nil
+
+	// try other typical errors
+	s.Run("it returns false on other errors", func() {
+		_, err = s.db.Create("users", testUser{ID: createdUser[0].ID})
+		s.Require().NotNil(err)
+		s.False(surrealdb.IsDuplicateUniqueIdx(err))
+
+		err = nil
+
+		_, err = s.db.Select("users:notexists")
+		s.Equal(err, surrealdb.ErrNoRow)
+		s.False(surrealdb.IsDuplicateUniqueIdx(err))
+	})
+	err = nil
+
+	// and finally, check with a tricky ID; this is the only way the function can give a false positive
```

Comment:
Can we change this to an ignored testcase? That's because we want to verify important behaviour rather than verify implementation. Also it's also ok to have to a test in the comments rather than a working test.